docker/Dockerfile

```diff
@@ -0,0 +1,25 @@
+#last version of ubuntu with blktap-utils
+FROM ubuntu:xenial
+
+# https://qastack.fr/programming/25899912/how-to-install-nvm-in-docker
+
+RUN apt-get update
+RUN apt-get install -y curl qemu-utils blktap-utils vmdk-stream-converter git
+ENV NVM_DIR /usr/local/nvm
+RUN mkdir -p /usr/local/nvm
+RUN cd /usr/local/nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
+ENV NODE_VERSION v17.0.1
+RUN /bin/bash -c "source $NVM_DIR/nvm.sh && nvm install $NODE_VERSION && nvm use --delete-prefix $NODE_VERSION"
+
+ENV NODE_PATH $NVM_DIR/versions/node/$NODE_VERSION/lib/node_modules
+ENV PATH      $NVM_DIR/versions/node/$NODE_VERSION/bin:$PATH
+
+RUN  npm install -g yarn
+
+WORKDIR /xen-orchestra
+#  invalidate build on package change
+COPY ./package.json /xen-orchestra/package.json
+COPY ./yarn.lock /xen-orchestra/yarn.lock
+ENTRYPOINT yarn ci
```

Comment:
From what you told me, we may need to add a `RUN yarn` before.

Reply:
I tested it, and this line seems required to ensure correct Layer invalidation and caching by satacket/action-docker-caching